Sources/swift-build-prebuilts/main.swift

```diff
@@ -0,0 +1,437 @@
+        self.components = components
+    }
+
+    init(_ value: String) {
+#if os(Windows)
+        let winValue = value.replacingOccurrences(of: "/", with: "\\")
+        let driveComps = winValue.split(separator: ":")
+        if driveComps.count > 1 {
+            self.drive = driveComps[0]
+            self.components = driveComps[1].split(separator: "\\")
+            return
+        }
+#endif
+        self.drive = nil
+        self.components = value.split(separator: Self.separator)
+    }
+
+    var path: String {
+#if os(Windows)
+        if let drive {
+            return drive + ":\\" + components.joined(separator: "\\")
+        }
+#endif
+        return Self.separator + components.joined(separator: Self.separator)
+    }
+
+    var description: String {
+        path
+    }
+
+    static func /(lhs: Path, rhs: String) -> Path {
+        .init(drive: lhs.drive, components: lhs.components + [rhs[...]])
+    }
+
+    static func /(lhs: Path, rhs: [String]) -> Path {
+        .init(drive: lhs.drive, components: lhs.components + rhs.map({ $0[...] }))
+    }
+
+    var basename: Substring {
+        components.last ?? Self.separator[...]
+    }
+
+    var exists: Bool {
+        fm.fileExists(atPath: path)
+    }
+
+    func remove() throws {
+        guard exists else {
+            // Already gone
+            return
+        }
+        try fm.removeItem(atPath: path)
+    }
+
+    func mkdirs() throws {
+        try fm.createDirectory(atPath: path, withIntermediateDirectories: true)
+    }
+
+    func cd() {
+        _ = fm.changeCurrentDirectoryPath(path)
+    }
+
+    func cp(to other: Path) throws {
+        try fm.copyItem(atPath: path, toPath: other.path)
+    }
+
+    func ls() throws -> [String] {
+        try fm.contentsOfDirectory(atPath: path)
+    }
+
+    func compress(_ files: [String]) throws {
+#if os(Windows)
+        try shell("tar -acf \(path) \(files.joined(separator: " "))")
+#else
+        try shell("zip -r \(path) \(files.joined(separator: " "))")
+#endif
+    }
+
+    var url: URL {
+        URL(fileURLWithPath: path)
+    }
+
+    func checksum() throws -> String {
+        let contents = try ByteString(Data(contentsOf: self.url))
+        return SHA256().hash(contents).hexadecimalRepresentation
+    }
+
+    func read() throws -> Data {
+        try Data(contentsOf: url)
+    }
+
+    func write(_ data: Data) throws {
+        try data.write(to: url)
+    }
+}
+
+extension Workspace.PrebuiltsManifest.Platform {
+    var dockerTag: String? {
+        guard docker else {
+            return nil
+        }
+
+        switch self {
+        case .ubuntu_jammy_aarch64, .ubuntu_jammy_x86_64:
+            return "jammy"
+        case .ubuntu_focal_aarch64, .ubuntu_focal_x86_64:
+            return "focal"
+        case .rhel_ubi9_aarch64, .rhel_ubi9_x86_64:
+            return "rhel-ubi9"
+        case .amazonlinux2_aarch64, .amazonlinux2_x86_64:
+            return "amazonlinux2"
+        default:
```

Comment:
Windows maps to `windowsservercore-ltsc2022`

Reply:
This script doesn't build Windows using docker. It needs to be done self hosted.

Reply:
Why does it _need_ to be done self hosted instead of using Docker?

Reply:
True, that could be a thing.

Reply:
I'll add this in when I do the more complete pass of docker images after the break. I need to get docker set up on my Windows box.

Reply:
Hopefully that's an Intel box, as containers don't yet work on Apple silicon at least (they require Hyper-V which in turn requires nested virtualization, which isn't supported). Not sure if they even work on non-Apple ARM hardware either.

Reply:
Yes, we have some old Intel iron here in the office that I'll grab for my Windows machine so I can spend more time immersing myself in Windows.